cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1372,6 +1374,75 @@ func maybeMountSave(disk disks.Disk, rootdir string, encrypted bool, mountOpts *
 	return true, nil
 }
 
+func generateMountsModeRunCloudimgRootfsDraft1(mst *initramfsMountsState) error {
+	// 1. mount ESP
+	// TODO cpc images should maybe set partlabel on the UEFI parition
+	if err := mountPartitionMatchingKernelDisk(boot.InitramfsUbuntuSeedDir, "UEFI"); err != nil {
+		return err
+	}
+
+	// get the disk that we mounted the ESP from as a reference
+	// point for future mounts
+	disk, err := disks.DiskFromMountPoint(boot.InitramfsUbuntuSeedDir, nil)
+	if err != nil {
+		return err
+	}
+
+	// 3.1. measure model
+	err = stampedAction("run-model-measured", func() error {
+		return secbootMeasureSnapModelWhenPossible(mst.UnverifiedBootModel)
+	})
+	if err != nil {
+		logger.Noticef("ignoring failure to measure model: %v", err)
+	}
+	// at this point on a system with TPM-based encryption
+	// data can be open only if the measured model matches the actual
+	// run model.
+	// TODO:UC20: on ARM systems and no TPM with encryption
+	// we need other ways to make sure that the disk is opened
+	// and we continue booting only for expected models
+
+	// 3.2. mount Data
+	// Changing to SeedEncryptionKeyDir
+	runModeKey := filepath.Join(boot.InitramfsSeedEncryptionKeyDir, "cloudimg-rootfs.sealed-key")
+	opts := &secboot.UnlockVolumeUsingSealedKeyOptions{
+		AllowRecoveryKey: true,
+	}
+	unlockRes, err := secbootUnlockVolumeUsingSealedKeyIfEncrypted(disk, "cloudimg-rootfs", runModeKey, opts)
```

Comment:
At the moment, since I am still prototyping I'm trying to make things work as optionally encrypted. I think we will remove optionality once this goes more production ready.

Reply:
The images can be encrypted or not. and we want to support both.